### ISSUE TITLE
(s-exp -> *) upgrades

### ISSUE THREAD

**Issue by schanzer** [bootstrapworld/curriculum#388]
- [x] Our conversion is naive, to the extent that it forces parens. `3 + 2 * 6` does not need parens in arithmetic, but we have no way of outputting this unless we use a string or write straight mathjax. Both solutions result in a loss of maintainability. We may need to implement some Order-of-Operations logic in the converter, which (optionally) strips out parens when they are not necessary.
- [x] Right now the semantics of `BSLeaveAHoleHere` are compiled away in the finished output, so that there is no way to look at a solution and see from the output which parts a student had to fill in. We should preserve these as classNames in both `sexp->code` and `sexp->coe`.
- [ ] Similarly, `sexp->math` compiles away parts a student should fill in (@flannery-denny confirms that [there is a way to make elements of a mathjax expression bold](https://tex.stackexchange.com/questions/595/how-can-i-get-bold-math-symbols)). We should preserve these as classNames.
- [x] if `pi` shows up in the sexp, the math output should be the pi symbol. (in sexp->code for Pyret, the symbol should show as `num-pi`)

All of the above could be implemented as modifications and/or hacks on our current system. However...

- [x] We currently have no way of specifying more sophisticated math (e.g - a complex fraction like `(1 +6) / (9 - 2)`, or a radical that spans an entire expression like in our distance formula output. As a result, we are severely limited in terms of the math exercises we can generate. 

**Proposal**
Since our "ground truth" is actually math, perhaps s-expressions are just too limited. [Math.js](https://mathjs.org/) is a JS library that has a very nice (and customizable!) parser, which would (1) allow us to write very sophisticated arithmetic expressions in a mostly-natural way, (2) generate LaTeX that is rendered properly by MathJax, (3) generate s-expressions that we feed to our current pipeline. 

Math.js is _massive_, and includes code for evaluation, transformation, big numbers, complex numbers, matrices, etc. Fortunately, we can generate a JS blob [that has only the code we need](https://mathjs.org/docs/custom_bundling.html), so this shouldn't require blowing our `js-delay` timeout.

**Comment by schanzer** [comment]
UPDATE - @ds26gte convinced us to stick with s-expression for now, and he's going to upgrade them to cover most (but not all!) of the items listed above

**Comment by schanzer** [comment]
@ds26gte  is a wizard!
